cmd/notary/delegations.go

```diff
@@ -188,6 +193,11 @@ func (d *delegationCommander) delegationAdd(cmd *cobra.Command, args []string) e
 		}
 	}
 
+	// If the user passes --all-paths, give the "" path
+	if allPaths {
```

Comment:
What if we special case this to remove all the current paths when `""` gets added?